src/main/java/io/github/fixitlater/quizapi/QuestionController.java

```diff
@@ -24,29 +28,62 @@ public QuestionController(QuestionService questionService) {
 
     @GetMapping("/random")
     public ResponseEntity<QuestionWithAnswersDTO> getRandomQuestion(ServletRequest request,
-                                                                    @RequestParam(value="category", required = false) String category) {
+                                    @RequestParam(value="category", required = false) String category,
+                                    @RequestParam(value="lang", required = false) String language) {
         QuestionWithAnswersDTO questionWithAnswersDTO;
         Map<String, String[]> paramMap = request.getParameterMap();
-        if(!paramMap.containsKey("category")) {
-            questionWithAnswersDTO = questionService.getRandomQuestion();
-        } else {
-            questionWithAnswersDTO = questionService.getRandomQuestionFromCategory(category);
+        Language languageEnum;
+        Category categoryEnum;
+        String categoryNotNull;
+        if(!paramMap.containsKey("category")) category = "ANY";
+        try {
+            categoryEnum = Category.valueOf(category.toUpperCase());
+        } catch (IllegalArgumentException e){
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```

Comment:
https://www.baeldung.com/exception-handling-for-rest-with-spring

Można w wolnej chwili zrefactorować to i zaimplementować tę obsługę raz, a w Controllerach rzucać tylko dany wyjątek.